evaluate_discrimination.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+"""This script evaluates the discriminative task: each candidate model applies
+a probability to candidate continuations for a given primer. The higher the
+probability given to the true continuation, and lower the probability given to
+the false continuation(s), the better the score. In the 2019 competition, only
+two continuations are scored, but this script is written in such a way that we
+could accept an arbitrary number of false continuation scores.
+
+The input data files are expected to be of the format:
+    id,A,B,...,Z
+    01234abcd,0.00092,0.001,...,0.9
+    12345bcd3,0.092,0.01,...,0.7
+    ...
+
+Where id represents the identity of the exerpt being evaluated, the final
+column is the true continuation score, and all other columns are false
+continuation scores.
+"""
+import argparse
+import numpy as np
+import pandas as pd
+
+
+def softmax(x, axis=-1):
+    """Returns the softmax for each row of the input x. For numerical
+    stability, we subtract the maximum value from each row first before getting
+    the exponents.
+    """
+    exp = np.exp(x - np.max(x, axis=axis, keepdims=True))
+    return exp / np.sum(exp, axis=axis, keepdims=True)
+
+
+def negloglike(x, labels):
+    """Returns the negative log likelihood from each row of x (unnormalized
+    probabilities - we apply a softmax to each row first).
+    """
+    probs = softmax(x)
+    return -np.log(probs[:, labels])  # select only the prob of true lab
+    
+
+def get_scores(x, labels=None):
+    """Returns the required scores: accuracy, mean negative log likelihood (the
+    crossentropy), and the variance of the negative log likelihood. If labels
+    is not supplied (a list of indices indicating the true label for each row),
+    it is assumed the final column of each row is the true label (this is very
+    specific to our particular expected input!)"""
+    nr_obs = x.shape[0]
+    nr_cols = x.shape[1]
+    max_idx = nr_cols - 1
+    if labels is None:
+        labels = max_idx * np.ones(nr_obs, dtype=int)
+    accuracy = np.mean(np.argmax(x, axis=-1) == labels)
+    nll = negloglike(x, labels)
+    crossent = np.mean(nll)
+    nll_var = np.var(nll)
+    return nr_obs, accuracy, crossent, nll_var
+
+
+if __name__ == '__main__':
+    # TODO: Parse args
+    
+    # TODO: Get files to score
+    files = ['./mono.csv', './poly.csv']
+    
```

Comment:
The files should also come from the config, ideally. Locally, I'm using a separate config for mono and poly.